mso/resource_mso_tenant_policies_dhcp_relay_policy.go

```diff
@@ -0,0 +1,257 @@
+package mso
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/ciscoecosystem/mso-go-client/client"
+	"github.com/ciscoecosystem/mso-go-client/container"
+	"github.com/ciscoecosystem/mso-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceMSOTenantPoliciesDHCPRelayPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMSOTenantPoliciesDHCPRelayPolicyCreate,
+		Read:   resourceMSOTenantPoliciesDHCPRelayPolicyRead,
+		Update: resourceMSOTenantPoliciesDHCPRelayPolicyUpdate,
+		Delete: resourceMSOTenantPoliciesDHCPRelayPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceMSOTenantPoliciesDHCPRelayPolicyImport,
+		},
+
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"template_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 64),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"uuid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"providers": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"dhcp_server_address": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.IsIPAddress,
+						},
+						"application_epg": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"external_epg": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"dhcp_server_vrf_preference": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+		CustomizeDiff: func(d *schema.ResourceDiff, v interface{}) error {
+			for index, provider := range d.Get("providers").(*schema.Set).List() {
+				providerMap := provider.(map[string]interface{})
+				appEPG, appEPGExists := providerMap["application_epg"].(string)
+				extEPG, extEPGExists := providerMap["external_epg"].(string)
+
+				if appEPGExists && extEPGExists && appEPG != "" && extEPG != "" {
+					return fmt.Errorf("only one of 'application_epg' or 'external_epg' can be set for a provider at index position: %d", index)
+				}
```

Comment:
should we loop through entire list and provide back all indexes with an error?

Reply:
Yes, we can do that.